dbms/src/main/java/org/polypheny/db/ddl/DdlManagerImpl.java

```diff
@@ -2239,19 +2286,22 @@ public void dropCollection( LogicalCollection collection, Statement statement )
         Snapshot snapshot = catalog.getSnapshot();
 
         AdapterManager manager = AdapterManager.getInstance();
+        boolean isSource = false;
 
         List<AllocationEntity> allocations = snapshot.alloc().getFromLogical( collection.id );
         for ( AllocationEntity allocation : allocations ) {
-            manager.getStore( allocation.adapterId ).orElseThrow().dropCollection( statement.getPrepareContext(), allocation.unwrapOrThrow( AllocationCollection.class ) );
-
+            if ( manager.isStore( allocation.adapterId ) ) {
+                manager.getStore( allocation.adapterId ).orElseThrow().dropCollection( statement.getPrepareContext(), allocation.unwrapOrThrow( AllocationCollection.class ) );
+            } else {
+                manager.getSource( allocation.adapterId ).orElseThrow().dropCollection( statement.getPrepareContext(), allocation.unwrapOrThrow( AllocationCollection.class ) );
+            }
```

Comment:
This can be simplified by using `getAdapter` instead of distinguishing between stores and sources.

Reply:
Thanks for the suggestion, I have changed it.